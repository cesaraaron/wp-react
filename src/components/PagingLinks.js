import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

export const PagingLinks = ({ total, isHome, activeIndex }) => (
  <div>
    {[...Array(total)].map((val, idx) => {
      const currentIndex = idx + 1
      const to = isHome && currentIndex === 1 ? '/' : `/p/${currentIndex}`

      return activeIndex === currentIndex ? (
        <span style={{ padding: '5px' }} key={currentIndex}>
          {currentIndex}
        </span>
      ) : (
        <Link style={{ padding: '5px' }} to={to} key={currentIndex}>
          {currentIndex}
        </Link>
      )
    })}
  </div>
)

PagingLinks.propTypes = {
  isHome: PropTypes.bool,
  total: PropTypes.number.isRequired,
  activeIndex: PropTypes.number.isRequired
}
