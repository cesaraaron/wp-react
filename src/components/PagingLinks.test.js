import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { PagingLinks } from './PagingLinks'

test('PagingLinks.js', () => {
  const component = renderer.create(
    <MemoryRouter>
      <PagingLinks total={5} activeIndex={2} />
    </MemoryRouter>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
