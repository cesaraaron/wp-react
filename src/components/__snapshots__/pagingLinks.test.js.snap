// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagingLinks.js 1`] = `
<div>
  <a
    href="/p/1"
    onClick={[Function]}
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    1
  </a>
  <span
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    2
  </span>
  <a
    href="/p/3"
    onClick={[Function]}
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    3
  </a>
  <a
    href="/p/4"
    onClick={[Function]}
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    4
  </a>
  <a
    href="/p/5"
    onClick={[Function]}
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    5
  </a>
</div>
`;
